truncate_client/src/lib.rs

```diff
@@ -0,0 +1,92 @@
+mod active_game;
+mod debug;
+mod game;
+mod game_client;
+mod lil_bits;
+mod theming;
+#[cfg(target_arch = "wasm32")]
+mod web_comms;
+
+use game_client::GameClient;
+
+#[cfg(target_arch = "wasm32")]
+use eframe::wasm_bindgen::{self, prelude::*};
+#[cfg(target_arch = "wasm32")]
+use eframe::web::AppRunnerRef;
+#[cfg(target_arch = "wasm32")]
+use futures::channel::mpsc;
+
+#[cfg(target_arch = "wasm32")]
+#[wasm_bindgen]
+pub struct WebHandle {
+    handle: AppRunnerRef,
+}
+
+#[cfg(target_arch = "wasm32")]
+#[wasm_bindgen]
+impl WebHandle {
+    #[wasm_bindgen]
+    pub fn stop_web(&self) -> Result<(), wasm_bindgen::JsValue> {
+        let mut app = self.handle.lock();
+        app.destroy()
+    }
+
+    #[wasm_bindgen]
+    pub fn set_some_content_from_javasript(&mut self, _some_data: &str) {
+        let _app = self.handle.lock().app_mut::<GameClient>();
+        // _app.data = some_data;
+    }
+}
+
+#[cfg(target_arch = "wasm32")]
+#[wasm_bindgen]
+pub fn init_wasm_hooks() {
```

Comment:
Haha `wasm_bindgen` is confusing at the best of times. It's `lib.rs` because this is our library target, i.e. what gets compiled when running `cargo build --lib` (instead of `main.rs` for the native version). 

The web_comms.rs file uses the [ws_stream_wasm](https://crates.io/crates/ws_stream_wasm) crate which uses [web_sys](https://rustwasm.github.io/wasm-bindgen/api/web_sys/) under the hood, which interacts with the [Browser Websocket API](https://developer.mozilla.org/en-US/docs/Web/API/WebSocket) directly. So the comms never go through JS at all, we're handling everything pretty much the same as the native implementation directly from Rust.

The JS<>WASM interop we're using is exclusively `start_separate()` (name stolen from egui) which gives egui the canvas to render to. Everything else is Rust